### PR TITLE
Dev loadtestservice microsoft.load test service 2021 11 01 preview pubpr

### PR DESCRIPTION
**Pull request by mjmadhu**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [x] new service onboarding
      - [ ] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [x] Other, please clarify
This RP is new into public APIs repo but it has been functioning into private preview since quite a while now.
We’re targeting to introduce this RP into public preview this week for which we’d like to add public preview APIs and generate SDKs asap.
RPSaaSMaster in private api repo - [azure-rest-api-specs-pr/specification/loadtestservice/resource-manager at RPSaaSMaster · Azure/azure-rest-api-specs-pr (github.com)](https://github.com/Azure/azure-rest-api-specs-pr/tree/RPSaaSMaster/specification/loadtestservice/resource-manager)
Manifest link - [MICROSOFT.LOADTESTSERVICE - Repos (visualstudio.com)](https://armmanifest.visualstudio.com/One/_git/Manifest?path=/MICROSOFT.LOADTESTSERVICE)

  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month. - Nov 2021
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month. - Nov 2021
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-11-01-preview/loadtestservice.json [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+          "x-ms-client-flatten": true
+        },
+        "identity": {
+          "$ref": "../../../../../common-types/resource-management/v3/managedidentity.json#/definitions/SystemAssignedServiceIdentity",
+          "description": "The type of identity used for the resource."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/TrackedResource"
```
Comment (raosuhas) [inline specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-11-01-preview/loadtestservice.json:393]: >TrackedResource [](http://example.com/codeflow?start=94&length=15)

Tracked resource needs to have a location at th top level .. In your examples I see that the location is at the incorrect level.

File: specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-11-01-preview/loadtestservice.json [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+        "location": {
+          "type": "string",
+          "description": "The geo-location where the resource lives"
+        },
+        "properties": {
+          "description": "Load Test resource properties",
+          "$ref": "#/definitions/LoadTestProperties",
+          "x-ms-client-flatten": true
+        }
+      }
```
Comment (raosuhas) [inline specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-11-01-preview/loadtestservice.json:418]: you cannot update location in a PATCH : [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#patch-resource](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#patch-resource)

File: specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-11-01-preview/loadtestservice.json [hunk 3]
```diff
@@ -0,0 +1,467 @@
...
+          "$ref": "../../../../../common-types/resource-management/v3/managedidentity.json#/definitions/SystemAssignedServiceIdentity",
+          "description": "The type of identity used for the resource."
+        },
+        "location": {
+          "type": "string",
+          "description": "The geo-location where the resource lives"
+        },
+        "properties": {
+          "description": "Load Test resource properties",
+          "$ref": "#/definitions/LoadTestProperties",
```
Comment (raosuhas) [inline specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-11-01-preview/loadtestservice.json:415]: >LoadTestProperties [](http://example.com/codeflow?start=33&length=18)

PATCH is for updating indivudual properties not for the whole object. For example it does not make sense to include readonly properties like dataPlaneURI in a patch .. Please check one of the other swaggers in this repo to see how other teams have implemented this. Note that only Tags is a must for including in the PATCH body.

File: specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-11-01-preview/examples/LoadTests_CreateOrUpdate.json [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    "loadTestName": "myLoadTest",
+    "LoadTestResource": {
+      "location": "westus",
+      "tags": {
+        "Team": "Dev Exp"
+      },
+      "properties": {
+        "description": "This is new load test resource"
+      }
+    }
```
Comment (raosuhas) [inline specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-11-01-preview/examples/LoadTests_CreateOrUpdate.json:15]: This looks like description and name is the only arguments that is required from the customer to create whis resource. Can you please expand on how would the customer interact with this resource ? Like can they do anything on the resource from the control plane other than CRUD ?


The reason we ask this is to please note that people use ARM templates to configure azure resources all the time. and this resource seems like  it is not configurable at all when using an ARM template.

File: specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-11-01-preview/loadtestservice.json (hunk 3 above)
Comment (raosuhas) [reply specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-11-01-preview/loadtestservice.json:415]: Check one of the other specifications in this repo to see how they have implemented the PATCH. also please read through : [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#patch-resource](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#patch-resource)

File: specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-11-01-preview/loadtestservice.json [hunk 5]
```diff
@@ -0,0 +1,467 @@
...
+          {
+            "$ref": "#/parameters/LoadTestNameParameter"
+          },
+          {
+            "in": "body",
+            "name": "ResourcePatchRequestBody",
+            "description": "LoadTest resource update data",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ResourcePatchRequestBody"
```
Comment (raosuhas) [inline specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-11-01-preview/loadtestservice.json:287]: >ResourcePatchRequestBody [](http://example.com/codeflow?start=37&length=24)

Please rename definition : "ResourcePatchRequestBody"to reflect that this is a LoadTest resourcePatch not a generic resource patch

File: specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-11-01-preview/examples/LoadTests_CreateOrUpdate.json (hunk 4 above)
Comment (mjmadhu) [reply specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-11-01-preview/examples/LoadTests_CreateOrUpdate.json:15]: @raosuhas , the customer can only perform CRUD operations on the resource as most of our business logic lies in data-plane (we'll be submitting data-plane apis too for review soon). We just need name, description and location from the customer to create the resource.

File: specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-11-01-preview/loadtestservice.json (hunk 3 above)
Comment (mjmadhu) [reply specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-11-01-preview/loadtestservice.json]: Done, Thanks!

File: specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-11-01-preview/loadtestservice.json (hunk 5 above)
Comment (mjmadhu) [reply specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-11-01-preview/loadtestservice.json]: Done, thanks!

File: specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-12-01-preview/loadtestservice.json [hunk 6]
```diff
@@ -0,0 +1,469 @@
...
+      "type": "object",
+      "properties": {
+        "tags": {
+          "description": "Resource tags.",
+          "type": "object"
+        },
+        "identity": {
+          "$ref": "../../../../../common-types/resource-management/v3/managedidentity.json#/definitions/SystemAssignedServiceIdentity",
+          "description": "The type of identity used for the resource."
+        },
```
Comment (raosuhas) [inline specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-12-01-preview/loadtestservice.json:408]: I dont think it is possible to update the identity with a PATCH. Please verify and update this.

Comment (mjmadhu) [reply specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-12-01-preview/loadtestservice.json:408]: @raosuhas , for already created resources we get PATCH call when user enables the MI. Please let us know if we're missing anything here.

File: specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-12-01-preview/examples/LoadTests_Delete.json [hunk 7]
```diff
@@ -0,0 +1,13 @@
+{
+  "parameters": {
+    "api-version": "2021-12-01-preview",
+    "subscriptionId": "00000000-0000-0000-0000-000000000000",
+    "resourceGroupName": "dummyrg",
+    "loadTestName": "myLoadTest"
+  },
+  "responses": {
+    "200": {},
+    "202": {},
```
Comment (raosuhas) [inline specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-12-01-preview/examples/LoadTests_Delete.json:10]: >202 [](http://example.com/codeflow?start=5&length=3)

do you accept 202 for delete ?

Comment (mjmadhu) [reply specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-12-01-preview/examples/LoadTests_Delete.json:10]: It's an LRO which can send 202 Accespted as API response.

File: specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-12-01-preview/loadtestservice.json [hunk 8]
```diff
@@ -0,0 +1,469 @@
...
+            "$ref": "#/parameters/LoadTestNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- resource deleted"
+          },
+          "202": {
+            "description": "Accepted -- LoadTest resource delete request accepted"
+          },
```
Comment (raosuhas) [inline specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-12-01-preview/loadtestservice.json:340]: Do you support Async deletes ?  Please include the "x-ms-long-running-operation" if you do. 

You can see other teams implemented this here : 
[https://github.com/Azure/azure-rest-api-specs/search?q=%22x-ms-long-running-operation%22](https://github.com/Azure/azure-rest-api-specs/search?q=%22x-ms-long-running-operation%22)

**Review by raosuhas — APPROVED** [review]
:shipit:

**Comment by mjmadhu** [comment]
Thanks @raosuhas for the review.
@changlong-liu Please review the PR. We'd request to merge this on the public preview date of 30th Nov.

File: specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-12-01-preview/loadtestservice.json (hunk 8 above)
Comment (mjmadhu) [reply specification/loadtestservice/resource-manager/Microsoft.LoadTestService/preview/2021-12-01-preview/loadtestservice.json:341]: done, thanks!

**Review by changlong-liu — APPROVED** [review]

**Pull request merged**